### PR TITLE
feat: Phase 3+4 — frontend tests, LRU cache, CSV/JSON export, CHANGELOG

### PR DESCRIPTION
**Pull request by kagawagao**
## Summary

Phase 3 (Frontend Testing) + Phase 4 (Functional Enhancements) from the v2.1.0 roadmap.

### Phase 3 — Frontend Testing (89 tests)
| Test File | Tests | Covers |
|-----------|-------|--------|
| FileUpload.test.ts | 22 | File type detection (magic bytes, binary, CJK, extensions) |
| AiPanel.test.ts | 21 | SSE event parsing (thinking/tool_call/tool_result/text/DONE) |
| App.test.ts | 33 | Filter logic + statistics (14 filter combos + all edge cases) |
| Export.test.ts | 13 | CSV/JSON generation (RFC 4180, BOM, escaping) |

### Phase 4 — Enhancements
- **LRU Cache**: ToolResultCache for lazy-log tools (128 entries, 60s TTL, mtime-aware)
- **Scan Optimization**: scan_file_meta max_scan_lines early exit (autoPath uses 50K limit)
- **CSV/JSON Export**: Download buttons in LogViewer toolbar (client-side generation)
- **CHANGELOG**: v2.1.0 section with all changes

### Total: 181 tests (89 frontend + 92 backend)
- 🟢 Frontend: 89/89
- 🟢 Backend: 92/92

## Summary by CodeRabbit

* **New Features**
  * CSV and JSON export buttons for logs (RFC4180, UTF-8 BOM, timestamped filenames); disabled when no data.
  * Direct in-UI rendering of AI tool results.

* **Improvements**
  * LRU cache with TTL to speed repeated AI tool queries.
  * Early-termination for large file scans to avoid full reads.
  * Added i18n strings for export/no-data messaging.

* **Tests**
  * Expanded frontend/backend test suites using Vitest + Testing Library.

### CODE REVIEW

**Comment by kagawagao** [comment]
@copilot fix ci

File: backend/tests/test_lazy_log.py [hunk 1]
```diff
@@ -502,3 +570,272 @@ def test_agent_tools_logging(self, caplog):
...
+            result2 = _execute_lazy_log_tool("overview_local_log", {"max_lines": 5}, path)
+            import json
+
+            r1 = json.loads(result1)
+            r2 = json.loads(result2)
+            # They should differ since different max_lines
+            assert r1["total_lines"] != r2["total_lines"]
+        finally:
+            _lazy_tool_cache.clear()
+            os.unlink(path)
```
[the remark this answers is not in the document]
Comment (kagawagao) [reply backend/tests/test_lazy_log.py:765]: @copilot confirm this feedback
